sql/driver/conn.go

```diff
@@ -178,13 +178,18 @@ func (c *conn) Send(call sqlwire.Call) (*rows, error) {
 			} else if datum.IntVal != nil {
 				t[j] = *datum.IntVal
 			} else if datum.UintVal != nil {
-				t[j] = *datum.UintVal
+				// uint64 not supported by the driver.Value interface.
+				t[j] = int64(*datum.UintVal)
 			} else if datum.FloatVal != nil {
 				t[j] = *datum.FloatVal
 			} else if datum.BytesVal != nil {
 				t[j] = datum.BytesVal
 			} else if datum.StringVal != nil {
-				t[j] = *datum.StringVal
+				t[j] = []byte(*datum.StringVal)
+			}
+			if !driver.IsScanValue(t[j]) {
+				err := fmt.Errorf("Value of bad type %T returned by database", t[j])
```

Comment:
I agree that panicing due to data received from the outside would be bad. In this case, though, the panic would not be caused by the data received from the server, but from incorrect code in the driver (i.e. assigning a `string` or `uint64` to `driver.Value`).